src/type-layout.md

```diff
@@ -374,24 +445,32 @@ struct MyDFields;
 
 ### Primitive representations
 
+r[layout.repr.primitive]
+
+r[layout.repr.primitive.intro]
 The *primitive representations* are the representations with the same names as
 the primitive integer types. That is: `u8`, `u16`, `u32`, `u64`, `u128`,
 `usize`, `i8`, `i16`, `i32`, `i64`, `i128`, and `isize`.
 
+r[layout.repr.primitive.constraint]
 Primitive representations can only be applied to enumerations and have
 different behavior whether the enum has fields or no fields. It is an error
 for [zero-variant enums] to have a primitive representation. Combining
 two primitive representations together is an error.
 
 #### Primitive Representation of Field-less Enums
 
+r[layout.repr.primitive.enum]
+
```

Comment:
Throughout the changes here it seems to be inconsistent about whether or not a blank line is added after the rule.

I would say the predominate style seems to not include blank lines. Can we stay consistent and avoid those extra blank lines?

Reply:
The blank line here is because it's a section rule-id not a paragraph rule-id. There's no paragraph rule-id because the current text would only have an intro paragraph. This style (blank line after rules for sections, and omitting a lone `.intro` id) has been common throughout the PRs.

Reply:
Could the section rule-id go above the main header?

Reply:
OK, I'm still not sure I agree with this, since the paragraph below does not look like an introduction to me. It seems to define the specific layout of these types of enums. I'm not going to block here, but I have opened #1651 to further discuss how these things interact.